go/vt/vttablet/tabletmanager/rpc_replication.go

```diff
@@ -42,9 +42,6 @@ var setSuperReadOnly bool
 var disableReplicationManager bool
 
 func registerReplicationFlags(fs *pflag.FlagSet) {
-	fs.Bool("enable_semi_sync", false, "")
-	fs.MarkDeprecated("enable_semi_sync", "please set the correct durability policy on the keyspace instead.")
```

Comment:
We should remove this flag's usage in our end-to-end tests as well. The field `EnableSemiSync` can be removed from `LocalProcessCluster`.

I think this would unblock three of the failing CI workflows.

Reply:
I just realized that removing flags on main can break upgrade/downgrade tests on v15 - because we test with N+1. We may need to go back and check how this affects those tests on release-15.0 branch.

Reply:
I have remove `EnableSemiSync` from the vttablet process and the local process cluster. Also @deepthi is right, I'll have to fix upgrade-downgrade tests in v15 too. I'll do that in a separate PR.

Reply:
Sounds good